### DISCUSSION TITLE
<title> How do I solve this problem?

### DISCUSSION THREAD

**Discussion by QWER236478** [community/community#126543] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

![image](https://github.com/community/community/assets/164462713/2c951a9c-63a4-48ad-a8d9-f5c65d7b56f4)
![1111](https://github.com/community/community/assets/164462713/c7ebbd85-7081-4a99-9763-e85428e1e2d1)

I want to make the following in this state for the first time. What should I do? It's my first time studying, so it's very difficult. Please help me

**Comment by thevindu-w** [comment]
I guess you are currently in the state shown in the 1st image, and you need to get to the state shown in the 2nd image. If so, the following steps will do it. (I used the commit messages C2, C3, C4, C5, and C6 in the following steps. You may replace them with different commit messages)
1. make some changes to the repo (add/modify/delete files).
2. run the following commands
```
git add .
git commit -m 'C2'
```
3. run the following command
```
git branch bugfix
```
4. make some changes to the repo (add/modify/delete files) .
5. run the following commands
```
git add .
git commit -m 'C3'
```
6. run the following command
```
git checkout bugfix
```
7. make some changes to the repo (add/modify/delete files) . Make sure these changes will not conflict with the changes made in step 4. (i.e. you should not change the same line of the same file in step 4 and this step)
8. run the following commands
```
git add .
git commit -m 'C4'
```
9. run the following command
```
git merge --no-edit -m 'C5' main
```
10. make some changes to the repo (add/modify/delete files) .
11. run the following commands
```
git add .
git commit -m 'C6'
```

**Comment by davevad93** [comment]
Hi @QWER236478 ,

To do so, follow these steps below:

1.  `git commit`

![1](https://github.com/community/community/assets/94308234/77622a1b-c6b8-4a81-9528-7f02a5de0f6f)

2. `git commit`

![2](https://github.com/community/community/assets/94308234/6f41e175-547e-48f2-8a81-aafc03bf1f89)

3.  `git checkout -b bugfix HEAD~1`

![3](https://github.com/community/community/assets/94308234/08efd17c-be5e-455e-be40-292926ef16ae)

4. `git commit`

![4](https://github.com/community/community/assets/94308234/1b3e5e63-a344-4084-a3c7-a63c3469a99f)

5. `git merge main`

![5](https://github.com/community/community/assets/94308234/7f969881-b4f5-4113-8d55-f2815fbc3e69)

6. `git commit`

![6](https://github.com/community/community/assets/94308234/ee242fca-65d8-4adf-be6e-9cb428852558)

**Reply by ebndev** [reply]
Hey @QWER236478!

Did this answer above work for you? Please feel encouraged to mark it as answered if so or let us know if you have further questions 😄
